### PR TITLE
DOC: finish docstring with numpy style

### PR DESCRIPTION
**Pull request by chiahaoliu**
follow numpy style and made sure test passed after modification. Sphinx doc can be render properly too. Ready for review @pavoljuhas

### CODE REVIEW

**Comment by chiahaoliu** [comment]
closes #12

File: src/diffpy/structure/lattice.py [hunk 1]
```diff
@@ -93,6 +150,29 @@ def __init__(self, a=None, b=None, c=None,
             alpha=None, beta=None, gamma=None,
             baserot=None, base=None):
         """define new coordinate system, the default is Cartesian
+
+        Parameters
+        ----------
+        a : float, optional
+            Lattice constant of a-axis, default to None.
```
Comment (pavoljuhas) [inline src/diffpy/structure/lattice.py]: Change to:
```
a : float or Lattice, optional
    Unit cell length a.  When present, other cell parameters must be also specified.
    When of the `Lattice` type, produce a duplicate object.
```

File: src/diffpy/structure/lattice.py [hunk 2]
```diff
@@ -148,13 +228,29 @@ def setLatPar(self, a=None, b=None, c=None,
...
-                    are in degrees.
-        baserot  -- unit cell rotation, base = stdbase @ baserot
-
-        Note: parameters with value None will remain unchanged.
-
-        No return value.
+        Parameters
+        ----------
+        a : float, optional
+            Lattice constant of a-axis, default to None.
```
Comment (pavoljuhas) [inline src/diffpy/structure/lattice.py]: Drop *default to None*.  Here `None` means not specified, i.e., the associated lattice parameter is thus left unchanged.

```
a : float, optional
    New value for the cell length `a`.

File: src/diffpy/structure/lattice.py [hunk 3]
```diff
@@ -271,34 +370,75 @@ def setLatBase(self, base):
...
         return rv
 
 
     def cartesian(self, u):
         """Return Cartesian coordinates of a lattice vector.
+
+        Parameters
+        ----------
+        u : ndarray
+            A lattice vector being projected to.
```
Comment (pavoljuhas) [inline src/diffpy/structure/lattice.py]: `being projected to` is confusing
change to
```
    A lattice vector in fractional coordinates or an N-by-3 array of lattice vectors.
```

File: src/diffpy/structure/lattice.py [hunk 4]
```diff
@@ -271,34 +370,75 @@ def setLatBase(self, base):
...
         return rc
 
 
     def fractional(self, rc):
         """Return fractional coordinates of a Cartesian vector.
+
+        Parameters
+        ----------
+        rc : ndarray
+            A lattice vector being projected to.
```
Comment (pavoljuhas) [inline src/diffpy/structure/lattice.py]: Here `rc` is a vector in Cartesian coordinates or an N-by-3 array of Cartesian vectors.

**Comment by pavoljuhas** [comment]
Hi Tim - I found a couple of things that can be improved - see above.
Am I allowed to commit to this PR?

**Comment by chiahaoliu** [comment]
hey Pavol, I can modify my PR based on your comments, so that you don't have to spend time on this :)

**Comment by chiahaoliu** [comment]
@pavoljuhas ready for round 2

**Comment by pavoljuhas** [comment]
@chiahaoliu - please check my above updates to 93d99bd; the easiest is to look at them commit by commit.

The main change is that `__init__` docstring did not show in HTML file so I moved it to the class docstring.

**Comment by pavoljuhas** [comment]
@chiahaoliu - are you fine with the updated version?

**Comment by chiahaoliu** [comment]
@pavoljuhas yes, it looks quite good. I think I forgot to click "submit" last week.

Should we merge it?

**Comment by pavoljuhas** [comment]
@chiahaoliu - sure, let me take care of it.  I have one tiny fix for the python3 branch that I'd like to get in before the merge.  Thank you, P.

**Pull request merged**